packages/react/src/components/Modal/Modal.tsx

```diff
@@ -234,7 +240,40 @@ export interface ModalProps extends HTMLAttributes<HTMLDivElement> {
   slug?: ReactNode;
 }
 
-const Modal = React.forwardRef(function Modal(
+const Modal = React.forwardRef<HTMLDivElement, ModalProps>(
+  ({ open, ...props }, ref) => (
+    <ModalPresence open={open}>
+      <ModalDialog {...props} ref={ref} />
+    </ModalPresence>
+  )
+);
+
+type ModalPresenceContextType = {
+  presenceRef: RefObject<HTMLDivElement | null>;
+  isExiting: boolean;
+};
+const ModalPresenceContext = createContext<ModalPresenceContextType>(
+  undefined as unknown as ModalPresenceContextType
+);
```

Comment:
Any reason not to initialize the context with a valid value?

```suggestion
const ModalPresenceContext = createContext<ModalPresenceContextType>({
  presenceRef: { current: null },
  isExiting: false,
});
```

Reply:
Just a habit to have occurrences throw if not called within a provider. Will update.